### PR TITLE
pytorch: weight decay

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by AnirudhDagar** [comment]
@mli this has failed now but it shall pass when #910 is merged. This depends on a function stored by #910
